### PR TITLE
Update kyma Istio Operator docs

### PR DESCRIPTION
**Pull request by nataliasitko**
Changes proposed in this pull request:

- Add Kyma Istio Operator's documentation fetched from the `istio` repo

### CODE REVIEW

**Review by strekm — APPROVED** [review]
/meow

**Review by IwonaLanger — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ Out of the box, Kyma offers various functionalities, such as:
 - [Serverless](https://kyma-project.io/#/01-overview/serverless/README) development platform to run lightweight Functions in a cost-efficient and scalable way
 - [System connectivity](https://kyma-project.io/#/01-overview/application-connectivity/README) that provides endpoint to securely register Events and APIs of external applications
 - [Eventing](https://kyma-project.io/#/01-overview/eventing/README) that provides messaging channel to receive events, enrich them, and trigger business flows using Functions or services
-- [Service Mesh](https://kyma-project.io/#/01-overview/service-mesh/README) for service-to-service communication and proxying
+- [Kyma Istio Operator](https://kyma-project.io/#/istio/user/00-overview/README) for service-to-service communication and proxying
```
Comment (a-thaler) [inline README.md:19]:
```suggestion
- [Istio](https://kyma-project.io/#/istio/user/00-overview/README) for service-to-service communication and proxying
```
I think it is confusing to use the operator term here, people are looking for the list of components and I suggest to use "Istio"

File: docs/01-overview/_sidebar.md [hunk 2]
```diff
@@ -6,5 +6,5 @@
   * [Observability](/01-overview/observability/README.md)
   * [Serverless](/01-overview/serverless/README.md)
   * [Service Management](/01-overview/service-management/README.md)
-  * [Service Mesh](/01-overview/service-mesh/README.md)
   * [User Interfaces](/01-overview/ui/README.md)
+  * [Kyma Istio Operator](/istio/user/00-overview/README.md)
```
Comment (a-thaler) [inline docs/01-overview/_sidebar.md:10]:
```suggestion
  * [Istio](/istio/user/00-overview/README.md)
```
Again, in the listing of components/functionality I would suggest to simply use "Istio" to not confuse the user

File: docs/04-operation-guides/troubleshooting/README.md [hunk 3]
```diff
@@ -59,9 +59,10 @@ See the full list of Kyma troubleshooting guides:
 - [Functions failing to build on k3d](./serverless/svls-04-function-build-failing-on-k3d.md)
 - [Serverless periodically restarting](./serverless/svls-05-serverless-periodically-restarting.md)
 
-### Service Mesh
+### Kyma Istio Operator
```
Comment (a-thaler) [inline docs/04-operation-guides/troubleshooting/README.md:62]:
```suggestion
### Istio
```

File: docs/04-operation-guides/troubleshooting/_sidebar.md [hunk 4]
```diff
@@ -6,7 +6,7 @@
   * [Observability](observability/README.md)
   * [Security](security/README.md)
   * [Serverless](serverless/README.md)
-  * [Service Mesh](service-mesh/README.md)
+  * [Kyma Istio Operator](/istio/user/02-operation-guides/troubleshooting/README.md)
```
Comment (a-thaler) [inline docs/04-operation-guides/troubleshooting/_sidebar.md:9]:
```suggestion
  * [Istio](/istio/user/02-operation-guides/troubleshooting/README.md)
```

File: docs/01-overview/service-mesh/README.md [hunk 5]
```diff
@@ -2,134 +2,4 @@
...
-  - You [can't access a Kyma endpoint](../../04-operation-guides/troubleshooting/service-mesh/smsh-01-503-no-access.md)
-  - [Connection refused errors](../../04-operation-guides/troubleshooting/service-mesh/smsh-02-connection-refused.md) occur
-  - [Issues with Istio sidecar injection](../../04-operation-guides/troubleshooting/service-mesh/smsh-03-istio-no-sidecar.md) come up
-  - You have an [incompatible Istio sidecar version after Kyma upgrade](../../04-operation-guides/troubleshooting/service-mesh/smsh-04-istio-sidecar-version.md)
-
-- Analyze configuration details for:
-
-   - The [Istio chart](../../05-technical-reference/00-configuration-parameters/smsh-01-istio-chart.md)
-   - The [Istio Resources chart](../../05-technical-reference/00-configuration-parameters/smsh-02-istio-resources-chart.md)
+This content has been moved to: [What is Kyma Istio Operator?](https://kyma-project.io/#/istio/user/00-overview/README.md).
```
[the remark this answers is not in the document]
Comment (nataliasitko) [reply docs/01-overview/service-mesh/README.md]: I added the name "What is Istio in Kyma?". Is that okay?

File: docs/01-overview/application-connectivity/ac-01-application-connector.md [hunk 6]
```diff
@@ -10,7 +10,7 @@ The external solution you connect to Kyma using AC is represented as an Applicat
 
 Application Connector secures Eventing with a client certificate verified by the Istio Ingress Gateway in the [Compass scenario](./README.md).
 
->**NOTE:** When using AC, make sure to [enable automatic Istio sidecar proxy injection](../../04-operation-guides/operations/smsh-01-istio-enable-sidecar-injection.md). For more details, see [Default Istio setup in Kyma](../../01-overview/service-mesh/smsh-02-default-istio-setup-in-kyma.md).
+>**NOTE:** When using AC, make sure to [enable automatic Istio sidecar proxy injection](https://kyma-project.io/#/iistio/user/02-operation-guides/operations/02-20-enable-sidecar-injection). For more details, see [Default Istio setup in Kyma](https://kyma-project.io/#/istio/user/00-overview/00-40-overview-istio-setup).
```
Comment (IwonaLanger) [inline docs/01-overview/application-connectivity/ac-01-application-connector.md:13]:
```suggestion
>**NOTE:** When using AC, make sure to [enable automatic Istio sidecar proxy injection](https://kyma-project.io/#/istio/user/02-operation-guides/operations/02-20-enable-sidecar-injection). For more details, see [Default Istio setup in Kyma](https://kyma-project.io/#/istio/user/00-overview/00-40-overview-istio-setup).
```

File: docs/01-overview/service-mesh/README.md [hunk 7]
```diff
@@ -2,134 +2,4 @@
...
-  - You [can't access a Kyma endpoint](../../04-operation-guides/troubleshooting/service-mesh/smsh-01-503-no-access.md)
-  - [Connection refused errors](../../04-operation-guides/troubleshooting/service-mesh/smsh-02-connection-refused.md) occur
-  - [Issues with Istio sidecar injection](../../04-operation-guides/troubleshooting/service-mesh/smsh-03-istio-no-sidecar.md) come up
-  - You have an [incompatible Istio sidecar version after Kyma upgrade](../../04-operation-guides/troubleshooting/service-mesh/smsh-04-istio-sidecar-version.md)
-
-- Analyze configuration details for:
-
-   - The [Istio chart](../../05-technical-reference/00-configuration-parameters/smsh-01-istio-chart.md)
-   - The [Istio Resources chart](../../05-technical-reference/00-configuration-parameters/smsh-02-istio-resources-chart.md)
+This content has been moved to: [What is Istio in Kyma?](https://kyma-project.io/#/istio/user/00-overview/README.md).
```
Comment (IwonaLanger) [inline docs/01-overview/service-mesh/README.md:5]: I'd remove the question mark or the period from here as they render next to each other (?.).

**Review by IwonaLanger — APPROVED** [review]

**Review by a-thaler — APPROVED** [review]

**Pull request merged**